zstd/decoder.go

```diff
@@ -156,8 +156,14 @@ func (d *Decoder) Reset(r io.Reader) error {
 	if d.current.err == ErrDecoderClosed {
 		return d.current.err
 	}
+
+	d.drainOutput()
+
 	if r == nil {
-		return errors.New("nil Reader sent as input")
+		d.current.b = []byte{}
```

Comment:
This should already be handled by the `drainOutput`, so this should be an un-needed alloc.
```suggestion
```

Reply:
Removed in the fixup.